### PR TITLE
US-ShortMessage-02 : Display "Switch to an article" when the number of caracters exceeds 1300

### PR DESCRIPTION
**Pull request by Rdidmohamedelhadi**

### CODE REVIEW

File: webapp/portlet/src/main/webapp/activity-composer-app/extension.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 let activityComposerActions = null;
-
-export function getActivityComposerExtensions() {
+let activityComposerHintAction = [];
```
Comment (thomasdelhomenie) [inline webapp/portlet/src/main/webapp/activity-composer-app/extension.js]: Why do we need an array for 1 action ?

File: webapp/portlet/src/main/webapp/activity-composer-app/extension.js [hunk 2]
```diff
@@ -9,6 +10,19 @@ export function getActivityComposerExtensions() {
...
+export function getActivityComposerHintActionExtensions() {
+  activityComposerHintListActions = getExtensionsByType('activity-composer-hint-action');
+  if (activityComposerHintListActions.length > 0 && activityComposerHintAction.length === 0) {
+    activityComposerHintAction.push(activityComposerHintListActions[0]);
+  }
+  for (let i = 1; i < activityComposerHintListActions.length; i++) {
+    if (activityComposerHintAction.rank > activityComposerHintListActions[i].rank) {
+      activityComposerHintAction = activityComposerHintListActions[i];
+    }
+  }
```
Comment (thomasdelhomenie) [inline webapp/portlet/src/main/webapp/activity-composer-app/extension.js:21]: Can you explain the logic please ?

**Review by thomasdelhomenie — CHANGES_REQUESTED** [review]

File: webapp/portlet/src/main/webapp/activity-composer-app/extension.js (hunk 2 above)
Comment (Rdidmohamedelhadi) [reply webapp/portlet/src/main/webapp/activity-composer-app/extension.js:22]: To ensure that the action take the least rankType a message

Comment (Rdidmohamedelhadi) [reply webapp/portlet/src/main/webapp/activity-composer-app/extension.js:21]: To ensure that the action take the least rank

Comment (thomasdelhomenie) [reply webapp/portlet/src/main/webapp/activity-composer-app/extension.js:21]: I see you simplified the logic, but it does not take the rank in account anymore, right ?

**Review by thomasdelhomenie — DISMISSED** [review]

**Review by thomasdelhomenie — APPROVED** [review]

**Pull request merged**